crates/forge/tests/cli/soldeer.rs

```diff
@@ -0,0 +1,142 @@
+//! Contains various tests related to `forge soldeer`.
+
+use std::{ fs::{ self, OpenOptions }, path::Path };
+
+use foundry_test_utils::forgesoldeer;
+use std::io::Write;
+forgesoldeer!(install_dependency, |prj, cmd| {
+    let command = "install";
+    let dependency = "forge-std~v1.8.1";
+
+    cmd.arg("soldeer").args([command, dependency]);
+    cmd.execute();
+
+    // Making sure the path was created to the dependency and that foundry.toml exists
+    // meaning that the dependencies were installed correctly
+    let path_dep_forge = prj
+        .root()
+        .join("dependencies")
+        .join("forge-std-v1.8.1")
+        .join("foundry.toml");
+    assert_eq!(path_dep_forge.exists(), true);
+
+    // Making sure the lock contents are the right ones
+    let path_lock_file = prj.root().join("soldeer.lock");
+    let lock_contents =
+        r#"
+[[dependencies]]
+name = "forge-std"
+version = "v1.8.1"
+source = "https://soldeer-revisions.s3.amazonaws.com/forge-std/v1_8_1_23-03-2024_00:05:44_forge-std-v1.8.1.zip"
+checksum = "0f7cd44f5670c31a9646d4031e70c66321cd3ed6ebac3c7278e4e57e4e5c5bd0"
+"#;
+
+    let actual_lock_contents = read_file_to_string(&path_lock_file);
+    assert_eq!(lock_contents, actual_lock_contents);
+
+    // Making sure the foundry contents are the right ones
+    let foundry_file = prj.root().join("foundry.toml");
+    let foundry_contents =
+        r#"[profile.default]
+src = "src"
+out = "out"
+libs = ["lib"]
+
+# See more config options https://github.com/foundry-rs/foundry/blob/master/crates/config/README.md#all-options
+
+[dependencies]
+forge-std = { version = "v1.8.1", url = "https://soldeer-revisions.s3.amazonaws.com/forge-std/v1_8_1_23-03-2024_00:05:44_forge-std-v1.8.1.zip" }
```

Comment:
am I correct in assuming this means there is no resolution at all? i.e. I can't just set `forge-std = "v1.8.1"` and it will resolve it?

Reply:
i was thinking in case soldeer is down, this can still be used, you know in case of running the update but i guess i can just remove it, also the `v` is optional, soldeer does not care what we put there as long as it's in the db. will work on it.

Reply:
solved. i just need to remove the v from the db, now it defaults to soldeer if no url